Can't build EnergyPlus on Centos 6.6

@Myoldmopar @lefticus 

I've tried building EnergyPlus 8.6 on Centos 6.6.  The default compiler is gcc 4.4 which doesn't seem to support c++11.  I get compiler crashes using gcc 4.7 and 4.8.  When I tried to install clang 3.4 I get GLIBCXX_3.4.15, GLIBC_2.15 and GLIBC_2.14 not found.  What compiler should I use?

@mbadams5 I spoke too soon, when I tried to make a package with the fortran components I am getting this error.  Do you see this as well?

```
Run CPack packaging tool...
CPack: Create package using TBZ2
CPack: Install projects
CPack: - Run preinstall target for: EnergyPlus
CPack: - Install project: EnergyPlus
 +++ AutoDocs: Completed processing output variable audit: processed 3795 calls
 +++ AutoDocs: Completed processing example file summary: processed 565 example files
 +++ AutoDocs: Completed processing example file links summary: processed 565 example files
CMake Error at /usr/share/cmake-3.6/Modules/GetPrerequisites.cmake:824 (message):
  /usr/bin/ldd failed: 1

Call Stack (most recent call first):
  /home/dmacumbe/EnergyPlus/cmake/ProjectMacros.cmake:196 (get_prerequisites)
  /home/dmacumbe/EnergyPlus/build/src/ExpandObjects/cmake_install.cmake:48 (fixup_executable)
  /home/dmacumbe/EnergyPlus/build/cmake_install.cmake:444 (include)


CPack Error: Error when generating package: EnergyPlus
make: *** [package] Error 1
```

--- a/src/EnergyPlus/DataSurfaces.hh
+++ b/src/EnergyPlus/DataSurfaces.hh
@@ -555,14 +555,7 @@ namespace DataSurfaces {
 		{
 			auto const & v1 = a.vertices;
 			auto const & v2 = b.vertices;
-			assert( v1.size_bounded() );
-			assert( v1.conformable( v2 ) );
-			if ( v1.empty() ) return true;
-			size_type l( 0u );
-			for ( int i = 1, e = v2.u(); i <= e; ++i, ++l ) {
-				if ( ! ( v1[ l ] == v2( i ) ) ) return false;
-			}
-			return true;		
+			return eq( v1, v2 );		
 		}
 
 		// Inequality

--- a/src/EnergyPlus/DataSurfaces.hh
+++ b/src/EnergyPlus/DataSurfaces.hh
@@ -553,7 +553,16 @@ namespace DataSurfaces {
 		bool
 		operator ==( Surface2D const & a, Surface2D const & b )
 		{
-			return eq( a.vertices, b.vertices );
+			auto const & v1 = a.vertices;
+			auto const & v2 = b.vertices;
+			assert( v1.size_bounded() );
+			assert( v1.conformable( v2 ) );
+			if ( v1.empty() ) return true;
+			size_type l( 0u );
+			for ( int i = 1, e = v2.u(); i <= e; ++i, ++l ) {
+				if ( ! ( v1[ l ] == v2( i ) ) ) return false;
+			}
+			return true;		
 		}
 
 		// Inequality

--- a/src/EnergyPlus/SurfaceOctree.cc
+++ b/src/EnergyPlus/SurfaceOctree.cc
@@ -159,7 +159,7 @@ namespace EnergyPlus {
 	SurfaceOctreeCube::
 	valid() const
 	{
-		if ( ( ( l_.x <= c_.x ) && ( l_.y <= c_.y ) ) && ( ( c_.x <= u_.x ) && ( c_.y <= u_.y ) ) ) {
+		if ( ( ( l_.x <= c_.x ) && ( l_.y <= c_.y ) && ( l_.z <= c_.z ) ) && ( ( c_.x <= u_.x ) && ( c_.y <= u_.y ) && ( c_.z <= u_.z ) ) ) {
 			Real const tol2( std::max( std::max( ObjexxFCL::magnitude_squared( l_ ), ObjexxFCL::magnitude_squared( u_ ) ) * ( 4 * std::numeric_limits< Real >::epsilon() ), 2 * std::numeric_limits< Real >::min() ) );
 			if ( ObjexxFCL::distance_squared( c_, cen( l_, u_ ) ) <= tol2 ) {
 				Real const tol( std::max( std::sqrt( std::max( ObjexxFCL::magnitude_squared( l_ ), ObjexxFCL::magnitude_squared( u_ ) ) ) * ( 4 * std::numeric_limits< Real >::epsilon() ), 2 * std::numeric_limits< Real >::min() ) );
@@ -221,7 +221,7 @@ namespace EnergyPlus {
 		std::uint8_t const k( ctr.z <= c_.z ? 0 : 1 );
 		SurfaceOctreeCube * & cube = cubes_[ ( i << 2 ) + ( j << 1 ) + k ];
 		if ( cube != nullptr ) { // Candidate cube exists
-			if ( ( ( cube->l_.x <= sl.x ) && ( cube->l_.y <= sl.y ) ) && ( ( su.x <= cube->u_.x ) && ( su.y <= cube->u_.y ) ) ) { // Surface is contained in sub-cube
+			if ( ( ( cube->l_.x <= sl.x ) && ( cube->l_.y <= sl.y ) && ( cube->l_.z <= sl.z ) ) && ( ( su.x <= cube->u_.x ) && ( su.y <= cube->u_.y ) && ( su.z <= cube->u_.z ) ) ) { // Surface is contained in sub-cube
 				cube->add( surface );
 			} else { // Surface stays in this cube
 				surfaces_.push_back( &surface );
@@ -232,7 +232,7 @@ namespace EnergyPlus {
 			Real const z( k * h );
 			Vertex const l( l_.x + x, l_.y + y, l_.z + z );
 			Vertex const u( c_.x + x, c_.y + y, c_.z + z );
-			if ( ( ( l.x <= sl.x ) && ( l.y <= sl.y ) ) && ( ( su.x <= u.x ) && ( su.y <= u.y ) ) ) { // Surface is contained in sub-cube
+			if ( ( ( l.x <= sl.x ) && ( l.y <= sl.y ) && ( l.z <= sl.z ) ) && ( ( su.x <= u.x ) && ( su.y <= u.y ) && ( su.z <= u.z ) ) ) { // Surface is contained in sub-cube
 				cube = new SurfaceOctreeCube( d_ + 1, l, u, h );
 				cube->add( surface );
 			} else { // Surface stays in this cube

--- a/src/EnergyPlus/SurfaceOctree.cc
+++ b/src/EnergyPlus/SurfaceOctree.cc
@@ -159,7 +159,7 @@ namespace EnergyPlus {
 	SurfaceOctreeCube::
 	valid() const
 	{
-		if ( le( l_, c_ ) && le( c_, u_ ) ) {
+		if ( ( ( l_.x <= c_.x ) && ( l_.y <= c_.y ) ) && ( ( c_.x <= u_.x ) && ( c_.y <= u_.y ) ) ) {
 			Real const tol2( std::max( std::max( ObjexxFCL::magnitude_squared( l_ ), ObjexxFCL::magnitude_squared( u_ ) ) * ( 4 * std::numeric_limits< Real >::epsilon() ), 2 * std::numeric_limits< Real >::min() ) );
 			if ( ObjexxFCL::distance_squared( c_, cen( l_, u_ ) ) <= tol2 ) {
 				Real const tol( std::max( std::sqrt( std::max( ObjexxFCL::magnitude_squared( l_ ), ObjexxFCL::magnitude_squared( u_ ) ) ) * ( 4 * std::numeric_limits< Real >::epsilon() ), 2 * std::numeric_limits< Real >::min() ) );
@@ -221,7 +221,7 @@ namespace EnergyPlus {
 		std::uint8_t const k( ctr.z <= c_.z ? 0 : 1 );
 		SurfaceOctreeCube * & cube = cubes_[ ( i << 2 ) + ( j << 1 ) + k ];
 		if ( cube != nullptr ) { // Candidate cube exists
-			if ( le( cube->l_, sl ) && le( su, cube->u_ ) ) { // Surface is contained in sub-cube
+			if ( ( ( cube->l_.x <= sl.x ) && ( cube->l_.y <= sl.y ) ) && ( ( su.x <= cube->u_.x ) && ( su.y <= cube->u_.y ) ) ) { // Surface is contained in sub-cube
 				cube->add( surface );
 			} else { // Surface stays in this cube
 				surfaces_.push_back( &surface );
@@ -232,7 +232,7 @@ namespace EnergyPlus {
 			Real const z( k * h );
 			Vertex const l( l_.x + x, l_.y + y, l_.z + z );
 			Vertex const u( c_.x + x, c_.y + y, c_.z + z );
-			if ( le( l, sl ) && le( su, u ) ) { // Surface is contained in sub-cube
+			if ( ( ( l.x <= sl.x ) && ( l.y <= sl.y ) ) && ( ( su.x <= u.x ) && ( su.y <= u.y ) ) ) { // Surface is contained in sub-cube
 				cube = new SurfaceOctreeCube( d_ + 1, l, u, h );
 				cube->add( surface );
 			} else { // Surface stays in this cube

--- a/src/EnergyPlus/SurfaceOctree.hh
+++ b/src/EnergyPlus/SurfaceOctree.hh
@@ -291,7 +291,7 @@ public: // Methods
 			return ObjexxFCL::distance_squared( a, c_ ) <= r_;
 		} else { // Might pay to check if a or b in sphere first in some applications
 			Vertex const ac( c_ - a );
-			Real const projection_fac( dot( ac, ab ) / ab_mag_squared );
+			Real const projection_fac( ( ( ac.x * ab.x ) + ( ac.y * ab.y ) ) / ab_mag_squared );
 			if ( ( 0.0 <= projection_fac ) && ( projection_fac <= 1.0 ) ) { // Projected (closest) point is on ab segment
 				return ObjexxFCL::distance_squared( ac, projection_fac * ab ) <= r_;
 			} else { // Projection (closest) point is outside of ab segment: Intersects iff a or b are in sphere
@@ -307,7 +307,7 @@ public: // Methods
 		assert( std::abs( dir.mag_squared() - 1.0 ) < 4 * std::numeric_limits< Real >::epsilon() ); // Check unit vector
 		// Might pay to check if a in sphere first in some applications
 		Vertex const ac( c_ - a );
-		Real const projection_fac( dot( ac, dir ) );
+		Real const projection_fac( ( ac.x * dir.x ) + ( ac.y * dir.y ) );
 		if ( 0.0 <= projection_fac ) { // Projected (closest) point is on ray
 			return ObjexxFCL::distance_squared( ac, projection_fac * dir ) <= r_;
 		} else { // Projection (closest) point is outside of ray: Intersects iff a is in sphere
@@ -321,7 +321,7 @@ public: // Methods
 	{
 		assert( std::abs( dir.mag_squared() - 1.0 ) < 4 * std::numeric_limits< Real >::epsilon() ); // Check unit vector
 		Vertex const ac( c_ - a );
-		return ac.mag_squared() - ObjexxFCL::square( ObjexxFCL::dot( ac, dir ) ) <= r_;
+		return ac.mag_squared() - ObjexxFCL::square( ( ac.x * dir.x ) + ( ac.y * dir.y ) ) <= r_;
 	}
 
 	// Line Segment Intersects Cube?

--- a/src/EnergyPlus/SurfaceOctree.hh
+++ b/src/EnergyPlus/SurfaceOctree.hh
@@ -291,7 +291,7 @@ public: // Methods
 			return ObjexxFCL::distance_squared( a, c_ ) <= r_;
 		} else { // Might pay to check if a or b in sphere first in some applications
 			Vertex const ac( c_ - a );
-			Real const projection_fac( ( ( ac.x * ab.x ) + ( ac.y * ab.y ) ) / ab_mag_squared );
+			Real const projection_fac( ( ( ac.x * ab.x ) + ( ac.y * ab.y ) + ( ac.z * ab.z ) ) / ab_mag_squared );
 			if ( ( 0.0 <= projection_fac ) && ( projection_fac <= 1.0 ) ) { // Projected (closest) point is on ab segment
 				return ObjexxFCL::distance_squared( ac, projection_fac * ab ) <= r_;
 			} else { // Projection (closest) point is outside of ab segment: Intersects iff a or b are in sphere
@@ -307,7 +307,7 @@ public: // Methods
 		assert( std::abs( dir.mag_squared() - 1.0 ) < 4 * std::numeric_limits< Real >::epsilon() ); // Check unit vector
 		// Might pay to check if a in sphere first in some applications
 		Vertex const ac( c_ - a );
-		Real const projection_fac( ( ac.x * dir.x ) + ( ac.y * dir.y ) );
+		Real const projection_fac( ( ac.x * dir.x ) + ( ac.y * dir.y ) + ( ac.z * dir.z ) );
 		if ( 0.0 <= projection_fac ) { // Projected (closest) point is on ray
 			return ObjexxFCL::distance_squared( ac, projection_fac * dir ) <= r_;
 		} else { // Projection (closest) point is outside of ray: Intersects iff a is in sphere
@@ -321,7 +321,7 @@ public: // Methods
 	{
 		assert( std::abs( dir.mag_squared() - 1.0 ) < 4 * std::numeric_limits< Real >::epsilon() ); // Check unit vector
 		Vertex const ac( c_ - a );
-		return ac.mag_squared() - ObjexxFCL::square( ( ac.x * dir.x ) + ( ac.y * dir.y ) ) <= r_;
+		return ac.mag_squared() - ObjexxFCL::square( ( ac.x * dir.x ) + ( ac.y * dir.y ) + ( ac.z * dir.z ) ) <= r_;
 	}
 
 	// Line Segment Intersects Cube?

--- a/src/EnergyPlus/Vectors.cc
+++ b/src/EnergyPlus/Vectors.cc
@@ -320,7 +320,7 @@ namespace Vectors {
 
 	void
 	DetermineAzimuthAndTilt(
-		Array1S< Vector > Surf, // Surface Definition
+		Array1D< Vector > const & Surf, // Surface Definition
 		int const EP_UNUSED( NSides ), // Number of sides to surface
 		Real64 & Azimuth, // Outward Normal Azimuth Angle
 		Real64 & Tilt, // Tilt angle of surface
@@ -555,7 +555,7 @@ namespace Vectors {
 
 	void
 	CreateNewellAreaVector(
-		Array1S< Vector > const VList,
+		Array1D< Vector > const & VList,
 		int const NSides,
 		Vector & OutNewellAreaVector
 	)
@@ -618,7 +618,7 @@ namespace Vectors {
 
 	void
 	CreateNewellSurfaceNormalVector(
-		Array1S< Vector > const VList,
+		Array1D< Vector > const & VList,
 		int const NSides,
 		Vector & OutNewellSurfaceNormalVector
 	)

--- a/src/EnergyPlus/Vectors.hh
+++ b/src/EnergyPlus/Vectors.hh
@@ -62,6 +62,7 @@
 // ObjexxFCL Headers
 #include <ObjexxFCL/Array1A.hh>
 #include <ObjexxFCL/Array1S.hh>
+#include <ObjexxFCL/Array1D.hh>
 
 // EnergyPlus Headers
 #include <EnergyPlus.hh>
@@ -112,7 +113,7 @@ namespace Vectors {
 
 	void
 	DetermineAzimuthAndTilt(
-		Array1S< Vector > Surf, // Surface Definition
+		Array1D< Vector > const & Surf, // Surface Definition
 		int const NSides, // Number of sides to surface
 		Real64 & Azimuth, // Outward Normal Azimuth Angle
 		Real64 & Tilt, // Tilt angle of surface
@@ -139,14 +140,14 @@ namespace Vectors {
 
 	void
 	CreateNewellAreaVector(
-		Array1S< Vector > const VList,
+		Array1D< Vector > const & VList,
 		int const NSides,
 		Vector & OutNewellAreaVector
 	);
 
 	void
 	CreateNewellSurfaceNormalVector(
-		Array1S< Vector > const VList,
+		Array1D< Vector > const & VList,
 		int const NSides,
 		Vector & OutNewellSurfaceNormalVector
 	);

